File: src/main/java/com/sequenceiq/cloudbreak/service/DuplicateKeyValueException.java
package com.sequenceiq.cloudbreak.service;


import com.sequenceiq.cloudbreak.domain.APIResourceType;

public class DuplicateKeyValueException extends RuntimeException {
    private final APIResourceType resourceType;
    private final String value;

    public DuplicateKeyValueException(APIResourceType resourceType, String value) {
        this.resourceType = resourceType;
        this.value = value;
    }

    public DuplicateKeyValueException(APIResourceType resourceType, String value, Throwable cause) {
        super(cause);
        this.resourceType = resourceType;
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public APIResourceType getResourceType() {
        return resourceType;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/DuplicateKeyValueException.kt
package com.sequenceiq.cloudbreak.service


import com.sequenceiq.cloudbreak.common.type.APIResourceType

class DuplicateKeyValueException : RuntimeException {
    val resourceType: APIResourceType
    val value: String

    constructor(resourceType: APIResourceType, value: String) {
        this.resourceType = resourceType
        this.value = value
    }

    constructor(resourceType: APIResourceType, value: String, cause: Throwable) : super(cause) {
        this.resourceType = resourceType
        this.value = value
    }
}
